crypto/CMakeLists.txt

```diff
@@ -182,11 +186,14 @@ if(ARCH STREQUAL "x86_64")
     CRYPTO_ARCH_SOURCES
 
     chacha/chacha-x86_64.${ASM_EXT}
-    cipher_extra/aes128gcmsiv-x86_64.${ASM_EXT}
     cipher_extra/chacha20_poly1305_x86_64.${ASM_EXT}
-    hrss/asm/poly_rq_mul.S
     test/trampoline-x86_64.${ASM_EXT}
   )
+  if(NOT MY_ASSEMBLER_IS_TOO_OLD_FOR_AVX)
```

Comment:
Instead of this, we can just guard everything in those files with `MY_ASSEMBLER_IS_TOO_OLD_FOR_AVX`?

Reply:
we can, but I thought this way is more clear. If you strongly prefer the other way I'll update the PR?

Reply:
My own preference is to exclude it in the file. My thought is that if we add a second condition `C` where we should only exclude file `hrss/asm/poly_rq_mul.S` from the build. Then that would be harder to implement. While if we do the exclusion in the file, we can simply do `&&`s.

On the other hand, this pattern cannot be used with third-party code we pull in e.g. s2n-bignum...

Thoughts?

Reply:
I agree, so I changed how we handle internal files. For the third-party files I think it's better to simply exclude them from the build